.github/workflows/test-all.yml

```diff
@@ -206,8 +209,71 @@ jobs:
           PKG=${{ matrix.package }}
           gap -A --quitonbreak -r -c "SetInfoLevel(InfoPackageLoading, PACKAGE_WARNING);res:=TestPackage(\"$PKG\");FORCE_QUIT_GAP(res);"
 
+      - name: "Get job status"
+        id: job-status-default
+        if: ${{ always() }}
+        run: |
+          PKG="${{ matrix.package }}"
+          echo '::set-output name=status::${{ job.status }}'
+
       - name: "Run tests with OnlyNeeded"
+        if: ${{ always() }}
         timeout-minutes: 12
         run: |
           PKG=${{ matrix.package }}
           gap -A --quitonbreak -r -c "SetInfoLevel(InfoPackageLoading, PACKAGE_WARNING);res:=TestPackage(\"$PKG\":OnlyNeeded);FORCE_QUIT_GAP(res);"
+
+      - name: "Get job status with OnlyNeeded"
+        id: job-status-only-needed
+        if: ${{ always() }}
+        run: |
+          PKG="${{ matrix.package }}"
+          echo '::set-output name=status::${{ job.status }}'
+
+      - name: "Create job status json-file"
+        if: ${{ always() }}
+        run: |
+          PKG="${{ matrix.package }}"
+          STATUS_DEFAULT="${{ steps.job-status-default.outputs.status }}"
+          STATUS_ONLY_NEEDED="${{ steps.job-status-only-needed.outputs.status }}"
```

Comment:
To my understanding, these two variables will end up with string values like `success`, `failure`, `cancelled`, or `skipped`, right? Perhaps we should add a comment here stating that, possibly with a link to the relevant docs.

Reply:
I will add a comment there and a link to the documentation. With `steps.<step_id>.outcome` we get four states, with `job.status` we get three states, so I will need to adjust the scripts slightly.